extensions/extensionservice/org.eclipse.smarthome.extensionservice.marketplace/src/main/java/org/eclipse/smarthome/extensionservice/marketplace/internal/MarketplaceProxy.java

```diff
@@ -0,0 +1,99 @@
+/**
+ * Copyright (c) 2014-2017 by the respective copyright holders.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.smarthome.extensionservice.marketplace.internal;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.smarthome.config.xml.util.XmlDocumentReader;
+import org.eclipse.smarthome.core.common.ThreadPoolManager;
+import org.eclipse.smarthome.extensionservice.marketplace.internal.model.Marketplace;
+import org.eclipse.smarthome.extensionservice.marketplace.internal.model.Node;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This class provides local access to the market place content. Once started, it downloads the catalog and then makes
+ * its content available from memory.
+ *
+ * Note that there is no progressive/lazy browsing implemented yet, but the service downloads the whole catalog.
+ * Once the marketplace is filled with a lot of content, this will need to be addressed.
+ *
+ * @author Kai Kreuzer - Initial contribution and API
+ *
+ */
+public class MarketplaceProxy {
+
+    private final Logger logger = LoggerFactory.getLogger(MarketplaceProxy.class);
+
+    private final static String MP_URL = "https://marketplace.eclipse.org/taxonomy/term/4988%2C4396/api/p?client=org.eclipse.smarthome";
+    private final URL url;
+    private Node[] cachedNodes = null;
+    private long refresh_interval = 3600;
+    private long retry_delay = 60;
+    private ScheduledFuture<?> refreshJob;
+
+    /**
+     * Creates a new instance, which immediately schedules a synchronization with the marketplace content.
+     */
+    public MarketplaceProxy() {
+        try {
+            url = new URL(MP_URL);
+            this.refreshJob = Executors.newSingleThreadScheduledExecutor().scheduleWithFixedDelay(() -> refresh(), 0,
+                    refresh_interval, TimeUnit.SECONDS);
```

Comment:
IIRC you should call shutdown after you added the job (https://docs.oracle.com/javase/7/docs/api/java/util/concurrent/ExecutorService.html#shutdown()).
If you don't use it, the executor does not "know" about, that it does not accept any further jobs.

```Java
final ScheduledExecutorService executor = Executors.newSingleThreadScheduledExecutor();
this.refreshJob = executor.scheduleWithFixedDelay(() -> refresh(), 0, refresh_interval, TimeUnit.SECONDS);
executor.shutdown();
```